src/java.base/share/classes/java/time/chrono/HijrahChronology.java

```diff
@@ -1035,9 +1036,8 @@ private static void registerCustomChrono() {
         AccessController.doPrivileged(
             (PrivilegedAction<Void>)() -> {
                 if (Files.isDirectory(CONF_PATH)) {
-                    try {
-                        Files.list(CONF_PATH)
-                            .map(p -> p.getFileName().toString())
+                    try (Stream<Path> stream = Files.list(CONF_PATH)) {
+                        stream.map(p -> p.getFileName().toString())
                             .filter(fn -> fn.matches("hijrah-config-[^\\.]+\\.properties"))
```

Comment:
BTW Should this use RESOURCE_PREFIX/SUFFIX and is the \\ inside [character class] suppose to quote the .?

Reply:
@ecki this seems orthogonal to this particular issue. Maybe it could be considered separately, if needed.